src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/EcmaFormat/ReflectionTypeProvider.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
```

Comment:
This class looks like it was copied over from somewhere else? Do we really need all of it? There's some duplication of functionality between this and helpers already existing in Reflection.Core.